pkg/promlib/models/query.go

```diff
@@ -63,8 +63,11 @@ type PrometheusQueryProperties struct {
 	// Series name override or template. Ex. {{hostname}} will be replaced with label value for hostname
 	LegendFormat string `json:"legendFormat,omitempty"`
 
-	// ???
+	// A set of filters applied to apply to the query
 	Scope *ScopeSpec `json:"scope,omitempty"`
+
+	// Additional Ad-hoc filters that take precendence over Scope on conflict.
+	AdhocFilters []ScopeFilter `json:"adhocFilters,omitempty"`
```

Comment:
Not sure this is the right property name?

Reply:
it's `filters` (https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/datasource.ts#L548)

Reply:
I can change it, but for other datasources, it seems like there is a good chance that we will end up with a collision there, and the property will have to be different elsewhere. WDYT?

Reply:
We have `filters` but we need this info in each query. So I think `adhocFilters` is good.

Reply:
Good call @kylebrandt, sorry, I've confused this query type with `DataQueryRequest` which is data source agnostic.

I would probably name it `QueryFilters` to align with `ApplyQueryFilters` function, or the other way around, rename that function to `ApplyAdhocFilters`. But consider this a detail.